### PR TITLE
Add Dashboard Header & Update Header Styles - SEAB-5199/SEAB-5226

### PR DESCRIPTION
**Pull request by KimberleyChong**
**Description**
Adds a header to the new Dashboard, adds breadcrumbs to headers for public workflows, tools etc., and updates header styles to align with the mock-ups.

### My Dockstore Headers:
My Dockstore headers are now a slightly larger font, icons are removed, they are fully left justified and now start with `My Dockstore:`.

### Before:

https://user-images.githubusercontent.com/97123241/221301577-90e65aef-72f4-493f-8c67-8d2645e1d07a.mp4

### After:

https://user-images.githubusercontent.com/97123241/221301594-0310a40c-4bfa-4ac3-bba9-9daaffd4b153.mp4

**Mocks:**
[My-account](https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/602c31e39929991717e349a4), [My-starred](https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/602c2ef936236428127aa107), [My-tools](https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/6022f90accad542f52f5acfd), [My-workflows](https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/60219b1cf92a1d19c95a33a2), [My-dashboard](https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/60229885ad1715098caf03e5)

### Search Header & Breadcrumbs
The search page now says "Explore Workflows" or "Explore Tools" according to the chosen tab.  And when viewing public tools/workflows the new breadcrumbs include navigation back to the search page.  Handles long entry names with an ellipsis to prevent overflow when resizing.

### Before:

https://user-images.githubusercontent.com/97123241/221301641-2ce10663-bb3c-431c-9954-7094dcb32ab4.mp4

### After:

https://user-images.githubusercontent.com/97123241/221301648-a2862a07-9a48-4525-8ed9-58bac712b001.mp4

Mocks:
[Public workflows](https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/60219b1cf92a1d19c95a33a2), [public tools](https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/6021aea03d1a3115fd55b473), [workflow search](https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/5fc690e874e0c807afeb299b), [tool search](https://app.zeplin.io/project/5fc690a2717ee9144814f190/screen/5fc7ceb4fd8d340926d62d6d)

`src/app/workflows/workflows.component` handles headers for workflows, services, notebooks.
https://github.com/dockstore/dockstore-ui2/blob/bf46e87d9bbcf8b2d04cde4c026fab1bb2015c93/src/app/workflows/workflows.component.ts#L34

Workflows will have a link back to the `workflow` tab of the search page:
https://github.com/dockstore/dockstore-ui2/blob/bf46e87d9bbcf8b2d04cde4c026fab1bb2015c93/src/app/workflows/workflows.component.html#L21-L27

Services and notebooks will have a link back to either `/services` or `/notebooks` (since they do not exist on the search page... yet):
https://github.com/dockstore/dockstore-ui2/blob/bf46e87d9bbcf8b2d04cde4c026fab1bb2015c93/src/app/workflows/workflows.component.html#L29-L31

To support the notebooks header I updated the `urlResolver` that is used:
https://github.com/dockstore/dockstore-ui2/blob/bf46e87d9bbcf8b2d04cde4c026fab1bb2015c93/src/app/shared/url-resolver.service.ts#L47-L48

Here's a demo with services, notebooks are handled the same way so it should behave the same when ready:

https://user-images.githubusercontent.com/97123241/221302025-2927a414-2218-40f6-9233-a2f5c858b8f6.mp4

Note: I didn't add these new arrow buttons to the headers, I think for the work this would need a its own ticket and a discussion if we would like to implement it

**Review Instructions**

**Issue**
SEAB-5199 SEAB-5226

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that your code compiles by running `npm run build`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] If this is the first time you're submitting a PR or even if you just need a refresher, consider reviewing our [style guide](https://github.com/dockstore/dockstore/wiki/Dockstore-Frontend-Opinionated-Style-Guide#pr-checklist)
- [x] Do not bypass Angular sanitization (bypassSecurityTrustHtml, etc.), or justify why you need to do so
- [x] If displaying markdown, use the `markdown-wrapper` component, which does extra sanitization
- [x] Do not use cookies, although this may change in the future
- [x] Run `npm audit` and ensure you are not introducing new vulnerabilities
- [x] Do due diligence on new 3rd party libraries, checking for CVEs
- [x] Don't allow user-uploaded images to be served from the Dockstore domain
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.
- [x] Check whether this PR disables tests. If it legitimately needs to disable a test, create a new ticket to re-enable it in a specific milestone.

### CODE REVIEW

**Comment by KimberleyChong** [comment]
> [Bug](https://sonarcloud.io/project/issues?id=dockstore_dockstore-ui2&pullRequest=1718&resolved=false&types=BUG) [B](https://sonarcloud.io/project/issues?id=dockstore_dockstore-ui2&pullRequest=1718&resolved=false&types=BUG) [3 Bugs](https://sonarcloud.io/project/issues?id=dockstore_dockstore-ui2&pullRequest=1718&resolved=false&types=BUG)

Ah, need to add 3 `alt` tags to the new `imgs`, will wait to push changes with any pr feedback

File: src/app/workflows/workflows.component.html [hunk 1]
```diff
@@ -13,15 +13,33 @@
...
+
 <app-header>
-  <div fxLayout="row" fxLayoutAlign="start center">
+  <div fxLayout fxLayoutAlign="start center" *ngIf="searchPage === false; else search">
+    <img src="../../assets/svg/search.svg" class="site-icons-small mr-3" />
+    <div fxLayout fxLayoutGap="1rem">
+      <a
+        *ngIf="(entryType$ | async) === EntryType.BioWorkflow; else serviceOrNotebook"
+        [routerLink]="['/search']"
+        [queryParams]="{ entryType: workflows }"
```
Comment (coverbeck) [inline src/app/workflows/workflows.component.html:24]: I think `workflows` needs to be quoted -- my IDE is flagging it.

File: src/app/workflows/workflows.component.ts [hunk 2]
```diff
@@ -15,24 +15,38 @@
...
+  private searchPageUrls: string[] = ['/workflows', '/notebooks', '/apptools', '/services'];
+  public searchPage: boolean = false;
 
-  constructor(private sessionQuery: SessionQuery, private sessionService: SessionService, private route: ActivatedRoute) {
+  constructor(
+    private sessionQuery: SessionQuery,
+    private sessionService: SessionService,
+    private route: ActivatedRoute,
+    private router: Router,
+    protected urlResolverService: UrlResolverService
```
Comment (coverbeck) [inline src/app/workflows/workflows.component.ts:42]: Does this need to be protected? If not, make private.

File: src/app/myworkflows/my-workflow/my-workflow.component.html [hunk 3]
```diff
@@ -21,20 +21,12 @@
...
-          <img src="../assets/svg/sub-nav/workflow.svg" alt="workflow icon" />
-          <span>{{ myEntryPageTitle$ | async | titlecase }}</span>
-        </div>
-        <div *ngIf="(myEntryPageTitle$ | async | titlecase) === 'My Services'" fxLayoutGap="0.5rem">
-          <img src="../assets/svg/sub-nav/services.svg" alt="service icon" />
-          <span>{{ myEntryPageTitle$ | async | titlecase }}</span>
-          <span class="beta-label"> preview </span>
-        </div>
+        <span>{{ myEntryPageTitle$ | async | titlecase }}</span>
+        <span *ngIf="(myEntryPageTitle$ | async | titlecase) === 'My Dockstore: Services'" class="beta-label ml-1"> preview </span>
```
Comment (kathy-t) [inline src/app/myworkflows/my-workflow/my-workflow.component.html:29]: This `preview` label is different from the one that the services entry box uses on the new dashboard. Perhaps we should update this one to use the new preview label?

File: src/app/workflows/workflows.component.html (hunk 1 above)
Comment (KimberleyChong) [reply src/app/workflows/workflows.component.html:24]: Thanks!

File: src/styles.scss [hunk 4]
```diff
@@ -1241,7 +1245,8 @@ table th {
 }
 
 /* Don't have multi-line headers */
-.mat-header-row {
+.mat-header-row,
+.white-space-no-wrap {
```
Comment (kathy-t) [inline src/styles.scss:1249]: Does `.no-wrap` not work?

https://github.com/dockstore/dockstore-ui2/blob/5dbe9958aacb844b8fec7dd8501231722660c8d8/src/styles.scss#L987-L991

Comment (KimberleyChong) [reply src/styles.scss:1249]: It works but `text-overflow: ellipsis;`  and `overflow: hidden;` cause the link to be cut off:

image

Compared to just having `white-space: nowrap;`:

image

**Review by coverbeck — APPROVED** [review]

**Review by svonworl — APPROVED** [review]

File: src/app/workflows/workflows.component.html [hunk 5]
```diff
@@ -13,15 +13,33 @@
...
+      <span fxHide.lt-sm class="truncate-text-1">{{ entryName }}</span>
+    </div>
+  </div>
+  <ng-template #search>
     <img *ngIf="(entryType$ | async) === EntryType.BioWorkflow" src="../assets/svg/sub-nav/workflow.svg" alt="workflow icon" />
     <img *ngIf="(entryType$ | async) === EntryType.Service" src="../assets/svg/sub-nav/services.svg" alt="service icon" />
     <img *ngIf="(entryType$ | async) === EntryType.Notebook" src="../assets/svg/sub-nav/notebooks.svg" alt="notebook icon" />
-
-    <span>&nbsp;{{ entryPageTitle$ | async | titlecase }}&nbsp;</span>
+    <span class="mx-2">{{ entryPageTitle$ | async | titlecase }}</span>
```
Comment (kathy-t) [inline src/app/workflows/workflows.component.html:40]: preview label below can also be updated to use the new style

**Review by kathy-t — APPROVED** [review]

**Comment by KimberleyChong** [comment]
Updated remaining preview label styles:

/services:
image
/notebooks:
image
/sitemap:
image

**Comment by denis-yuen** [comment]
> Note: I didn't add these new arrow buttons to the headers, I think for the work this would need a its own ticket and a discussion if we would like to implement it

Noted and definitely a 1.15 thing

**Review by denis-yuen — APPROVED** [review]

**Pull request merged**